### DISCUSSION TITLE
What is the right way to make a crontab for a container?

### DISCUSSION THREAD

**Discussion by LordJuice1** [Jip-Hop/jailmaker#252] [Q&A]
I need to make a [conjob](https://github.com/nextcloud/all-in-one/discussions/2810) for my Nextcloud AIO container. It's not my first time making one. But, I'm not sure where to put one for a Docker container thats inside jailmaker. Do I put it at the base OS of Truenas and modify it some to go into the jailmaker docker shell, and then call for the container?

**Comment by Lockszmith-GH** [comment]
A coupld of options, each with it's pros and cons:

1. Create a systemd timer unit INSIDE the jail
2. Use TrueNAS's cron job UI to call a shell script that calls within the jail.

With Option # 1 - need to create systemd unit files, but it will be self contianed, only running when the jail is active.
With Option # 2 - easy scheduling in the UI as well as the ability to trigger on demand. Requires a script to make sure the jail is active and running.

**Comment by LordJuice1** [comment]
Option 2 sounds like the way to go. What do you think is the best way to do that?

**Reply by Lockszmith-GH** [reply]
Write a script and place it inside the jail.

Easierst method:
From TrueNAS SCALE's UI add a crontab that call `<fullpath>/jlmkr.py exec <jail-name> <path to script inside jail>`
This will fail if the jail is down.
I believe you will only get an email from TrueNAS SCALE if the jail is down, as exit codes from within the jail are unlikely to propagate back out (I'm guessing, as I've not tested this)

Alternatively, you can write a script that runs on TrueNAS SCALE's host side with all the tests before calling the inside script.
Remember: when the script returns a non-zero exit code, TrueNAS SCALE will take is a failure of the crontab, and will send out an email (if it's configured correctly)
